### DISCUSSION TITLE
Does ts-node support only "module": "CommonJS"?

### DISCUSSION THREAD

**Discussion by PavelTurk** [TypeStrong/ts-node#1794] [Q&A] [answered]
I have the following settings in tsconfig:

```
  "ts-node": {
    "esm": true,
    "experimentalResolver" : true,
    "compilerOptions": {
      "target": "es2017",
      "module": "CommonJS",
      "moduleResolution": "node",
      "esModuleInterop": true,
      "forceConsistentCasingInFileNames": true,
      "strict": false,
      "skipLibCheck": true
    }
  }
```

My library is `es2020` module as it is a angular 13 library. But if I set "module": "es2020" in "ts-node" then it simply doesn't work -> SyntaxError: Unexpected token 'export'. Could anyone say if I am right saying that ts-node (10.8.1) supports only "module": "CommonJS"?

**Comment by cspotcode** [comment, marked as answer]
ts-node supports all module modes, including the new `NodeNext`.  I suspect you forgot to set "type" in your package.json

For you, I recommend using new new `NodeNext` mode.

To understand better what's going on, I recommend starting here:
https://typestrong.org/ts-node/docs/imports
And to learn about NodeNext, start here:
https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#esm-nodejs

If you choose *not* to use NodeNext, your project will still work!  But remember that package.json "type" will be important no matter what mode you choose.

Remember, node's ESM support still has rough edges, so care must be taken in project configuration.
